### PR TITLE
Replace .iter().copied() on arrays with .into_iter()

### PR DESCRIPTION
**Pull request by inthar-raven**
ref #381

### CODE REVIEW

File: server/src/sim.rs [hunk 1]
```diff
@@ -283,9 +283,9 @@ impl Sim {
         populate_fresh_nodes(&mut self.graph);
 
         let mut fresh_voxel_data = vec![];
-        for fresh_node in fresh_nodes.iter().copied() {
+        for fresh_node in (&fresh_nodes).into_iter() {
```
Comment (Ralith) [inline server/src/sim.rs:286]: `fresh_nodes` is a `Vec`, not an array, so it's not `Copy`, so this change doesn't improve clarity or concision.

**Review by Ralith — CHANGES_REQUESTED** [review]
We'll also want to handle the various cases of `.iter().cloned()`, which seems to actually be the more common, functionally equivalent pattern.

**Comment by inthar-raven** [comment]
`.into_iter()` consumes the iterator, and that caused some problems.

**Review by Ralith — APPROVED** [review]
Thanks!

**Pull request merged**